nx/lib/nx.ex

```diff
@@ -7332,6 +7332,36 @@ defmodule Nx do
     {Enum.reverse(list1), Enum.reverse(list2), Enum.reverse(list3), Enum.reverse(list4)}
   end
 
+  @doc """
+  Stacks an `Enumerable` of `n` scalar tensors into an `{n}`-shaped tensor
+
+  ### Options
+
+    * `:name` - optional name for the added dimension. Defaults to an unnamed axis.
+
+  ### Examples
+
+      iex> Nx.stack([1, 2, 3])
+      #Nx.Tensor<
+        s64[3]
+        [1, 2, 3]
+      >
+
+      iex> Nx.stack([Nx.tensor(1), Nx.tensor(2)], name: :x)
+      #Nx.Tensor<
+        s64[x: 2]
+        [1, 2]
+      >
+  """
+  def stack(enumerable, opts \\ []) do
```

Comment:
Should I validate that is is an enumerable somehow?

Also, should this be delegated to individual backends? It seems to me that this belongs here, but I might have missed something

Reply:
I would validate it's a list here, I think that makes sense

Reply:
Technically this also works for MapSets or any other Enumerable which yields tensors when `Enum.map`ed

Reply:
Indeed but I think those use cases won't be common, so I would go with something like:

```suggestion
  def stack(tensors, opts \\ []) when is_list(tensors) do
```